apispec/run.go

```diff
@@ -99,6 +100,21 @@ func Run(args Args) error {
 		}
 	}
 
+	// Import information about production or staging environment
+	{
+		deploymentType, deploymentTags := deployment.GetDeploymentInfo()
+
+		// Override user type, but not CI.
+		if deploymentType != deployment.None &&
+			args.Tags[tags.XAkitaSource] == tags.UserSource {
```

Comment:
This looks different from what is in `apidump.go`. Is this intentional? If not, maybe we should factor this into its own method somewhere.

Reply:
It is intentional, because here Tags[tags.XAkitaSource] has been explicitly initialized to UserSource above.

But, I am not too happy with the result.  I think the right thin in apispec is:

* if CI and source not set, set to CI
* if production and source not set, set to production
* if source not set, set to user

Instead of trying to do it in the reverse order.

Reply:
Rewrote this entirely.